### ISSUE TITLE
Generated files can't contain properties

### ISSUE THREAD

**Issue by AustinRP** [frthjf/obsidian-zotero-sync-client#1]
First off, great plugin! I spent some time browsing Obsidian/Zotero integrations and this one is my favorite, so thanks for creating it. :) 

In this plugin's config, I coded a template that includes some Obsidian [properties](https://help.obsidian.md/Editing+and+formatting/Properties). However, Obsidian doesn't detect these properties because the generated file contains a "marker" (the zotero key) and a Zotero link at the top (the relevant piece of code from the plugin is [here](https://github.com/frthjf/obsidian-zotero-sync-client/blob/8f3ae4339d4501d7b147f92b602476f9aa610029/main.ts#L300)).

I think the simplest, non-breaking fix would be an optional setting to read/write the marker at the end of the note instead of the beginning.  A more robust approach might be to include the marker as another file property, but then we'd need some way for the plugin user to create their own properties. 

My availability is honestly shaky right now, but I'd be happy to work on this if you're open to pull requests!

**Comment by frthjf** [comment]
Hi @AustinRP,

Thanks for reaching out, I am glad to hear the plugin is useful to you!

That makes sense, I did not consider properties. The idea behind the marker is to be able to detect if a file has been generated by the plugin to avoid accidental modification of user files. However, [I haven't actually implemented](https://github.com/frthjf/obsidian-zotero-sync-client/blob/8f3ae4339d4501d7b147f92b602476f9aa610029/main.ts#L372) such a fail safe yet, so you are right, changing the marker position shouldn't break anything.

How about we make the marker available as part of the template (e.g. under `data.marker`) so users can place them freely (e.g. after the property header or at the end of the file etc.). If one doesn't add the marker in the template, it gets automatically prepended as before. So something like this:

```js
element.marker = this.getMarker(element)
let note = this.generateNote(element)
if (!note.includes(element.marker)) {
    note = element.marker + '\n\n' + note
}
// getMarker needs to be updated to not include the '\n\n'
```

Would that work for you? 

And sure, I am always open to PRs although given this is a minor thing I'd also be happy to do it, just let me know.

**Comment by frthjf** [comment]
This is now implemented in [v1.2.0](https://github.com/frthjf/obsidian-zotero-sync-client/releases/tag/1.2.0).
